src/gui_common/CompoundAmount.cs

```diff
@@ -166,7 +166,11 @@ private void UpdateLabel()
         }
 
         string numberPart;
-        if (UsePercentageDisplay)
+        if (!string.IsNullOrEmpty(compound!.Unit))
+        {
+            numberPart = Math.Round(amount) + " " + compound.Unit;
```

Comment:
I think this should use the translation system, a key like "VALUE_WITH_UNIT", which would be `{0} {1}` for English, should probably be used as some language might want to switch the order of those here.